backend/src/main/java/com/storageroom/security/config/SecurityConfiguration.java

```diff
@@ -0,0 +1,49 @@
+package com.storageroom.security.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import javax.sql.DataSource;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+    private DataSource dataSource;
+
+    @Autowired
+    public SecurityConfiguration(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.jdbcAuthentication()
+                .withDefaultSchema()
+                .dataSource(dataSource)
+                .withUser("user")
+                .password("{bcrypt}" + new BCryptPasswordEncoder().encode("password"))
+                .roles("USER")
+                .and()
+                .withUser("admin")
+                .password("{bcrypt}" + new BCryptPasswordEncoder().encode("admin"))
+                .roles("ADMIN");
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/h2-console/**").permitAll()
+                .antMatchers("/user").hasAnyRole("USER", "ADMIN")
+                .antMatchers("/admin").hasRole("ADMIN")
+                .anyRequest().authenticated()
+                .and()
+                .formLogin().permitAll();
```

Comment:
Co to robi? Mamy gdzieś formLogin?

Reply:
Nie mamy, to jest domyslna templatka szpringowa. No wystawia formLogin dostepne dla kazdego, nie musisz byc zalogowany, wbijasz na stronke i masz